src/overlays/actors/ovl_Boss_05/z_boss_05.c

```diff
@@ -159,107 +341,1343 @@ ActorInit Boss_05_InitVars = {
+                    this->drawDmgEffState = BIO_BABA_DRAW_DMGEFF_STATE_NONE;
+                    this->drawDmgEffAlpha = 0.0f;
+                }
+            } else {
+                Math_ApproachF(&this->drawDmgEffScale, 1.0f, 0.5f, 0.5f);
+            }
+            break;
+    }
+}
+
+static s8 sLimbIndexToLimbRotIndex[] = {
+    BIO_BABA_LILY_PAD_WITH_HEAD_LIMB_ROT_INDEX_NONE,            // BIO_DEKU_BABA_LILY_PAD_LIMB_NONE
+    BIO_BABA_LILY_PAD_WITH_HEAD_LIMB_ROT_INDEX_NONE,            // BIO_DEKU_BABA_LILY_PAD_LIMB_ROOTS
+    BIO_BABA_LILY_PAD_WITH_HEAD_LIMB_ROT_INDEX_UPPER_STEM,      // BIO_DEKU_BABA_LILY_PAD_LIMB_UPPER_STEM
+    BIO_BABA_LILY_PAD_WITH_HEAD_LIMB_ROT_INDEX_MIDDLE_STEM,     // BIO_DEKU_BABA_LILY_PAD_LIMB_MIDDLE_STEM
+    BIO_BABA_LILY_PAD_WITH_HEAD_LIMB_ROT_INDEX_LOWER_STEM,      // BIO_DEKU_BABA_LILY_PAD_LIMB_LOWER_STEM
+    BIO_BABA_LILY_PAD_WITH_HEAD_LIMB_ROT_INDEX_LEFT_UPPER_ARM,  // BIO_DEKU_BABA_LILY_PAD_LIMB_LEFT_UPPER_ARM
+    BIO_BABA_LILY_PAD_WITH_HEAD_LIMB_ROT_INDEX_LEFT_LOWER_ARM,  // BIO_DEKU_BABA_LILY_PAD_LIMB_LEFT_LOWER_ARM
+    BIO_BABA_LILY_PAD_WITH_HEAD_LIMB_ROT_INDEX_RIGHT_UPPER_ARM, // BIO_DEKU_BABA_LILY_PAD_LIMB_RIGHT_UPPER_ARM
+    BIO_BABA_LILY_PAD_WITH_HEAD_LIMB_ROT_INDEX_RIGHT_LOWER_ARM, // BIO_DEKU_BABA_LILY_PAD_LIMB_RIGHT_LOWER_ARM
+    BIO_BABA_LILY_PAD_WITH_HEAD_LIMB_ROT_INDEX_NONE,            // BIO_DEKU_BABA_LILY_PAD_LIMB_LEAF
+    BIO_BABA_LILY_PAD_WITH_HEAD_LIMB_ROT_INDEX_NONE,            // BIO_DEKU_BABA_LILY_PAD_LIMB_MAX
+};
+
+s32 Boss05_OverrideLimbDraw_LilyPadWithHead(PlayState* play, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot,
+                                            Actor* thisx) {
+    Boss05* this = THIS;
+    s8 limbRotIndex;
+
+    if (limbIndex == KREG(32)) {
+        if ((this->frameCounter % 4) == 0) {
+            *dList = NULL;
+        }
+
+        rot->x += KREG(33) * 0x100;
+        rot->y += KREG(34) * 0x100;
+        rot->z += KREG(35) * 0x100;
+    }
+
+    if ((BIO_BABA_GET_TYPE(&this->dyna.actor) == BIO_BABA_TYPE_NO_LEAF_LILY_PAD_WITH_HEAD) &&
+        (limbIndex == BIO_DEKU_BABA_LILY_PAD_LIMB_LEAF)) {
+        *dList = NULL;
+    }
+
+    if (limbIndex == BIO_DEKU_BABA_LILY_PAD_LIMB_UPPER_STEM) {
+        rot->x += this->lilyPadWithHeadStemRotX;
+    }
 
-extern ColliderJntSphElementInit D_809F1B2C[2];
-extern ColliderJntSphInit D_809F1B74;
-extern ColliderJntSphElementInit D_809F1B84[1];
-extern ColliderJntSphInit D_809F1BA8;
-extern ColliderJntSphElementInit D_809F1BB8[1];
-extern ColliderJntSphInit D_809F1BDC;
-extern DamageTable D_809F1C00;
-extern DamageTable D_809F1C20;
+    limbRotIndex = sLimbIndexToLimbRotIndex[limbIndex];
+    if (limbRotIndex > BIO_BABA_LILY_PAD_WITH_HEAD_LIMB_ROT_INDEX_NONE) {
+        rot->x = rot->x + this->lilyPadWithHeadLimbRot[limbRotIndex].x;
+        rot->y = rot->y + this->lilyPadWithHeadLimbRot[limbRotIndex].y;
+        rot->z = rot->z + this->lilyPadWithHeadLimbRot[limbRotIndex].z;
```

Comment:
You would think it would work like that, but trying this causes regalloc issues:
![image](https://github.com/zeldaret/mm/assets/12245827/b18eb4a4-b8cb-4397-961d-ec050bc85cb3)

Reply:
rip